src/test/kotlin/org/rust/ide/inspections/RsMainFunctionNotFoundInspectionTest.kt

```diff
@@ -15,6 +15,12 @@ class RsMainFunctionNotFoundInspectionTest : RsInspectionsTestBase(RsMainFunctio
         <error descr="`main` function not found in crate `test-package` [E0601]"> /*caret*/ </error>
     """)
 
+    fun `test do not show if an error exists`() = checkByFileTree("""
```

Comment:
Let's also add test where file contains psi error element but as a nondirect child to ensure that we show error in this case